### DISCUSSION TITLE
Show error message .onepanel/cli_config.yaml no such file or directory

### DISCUSSION THREAD

**Discussion by rogerhsu1121** [onepanelio/core#911] [Get Help]
I follow the step to install onepanel in my local MicroK8s  on Ubuntu 20.04., 
after I do Step 1: Install Onepanel#
Run the following command to initialize a params.yaml template for your provider:

I execute the follow command
opctl init --provider microk8s \
           --enable-metallb \
           --artifact-repository-provider abs

But I get the error message. .onepanel/cli_config.yaml no such file or directory. where can I find or create this file to continue the process?

xxx@CTS1-SW2-OptiPlex-3020:/$ opctl init --provider microk8s \
>    --enable-metallb \
>            --artifact-repository-provider abs
2021/04/08 09:10:56 Initializing...
2021/04/08 09:10:56 [error] creating default source config: open .onepanel/cli_config.yaml: no such file or directory
xxx@CTS1-SW2-OptiPlex-3020:/$ cd /usr/local/bin/

**Comment by Vafilor** [comment]
Hi there,

It looks like the CLI is having permission issues creating the file. Can you try running the command with sudo?

```bash
sudo opctl init --provider microk8s --enable-metallb --artifact-repository-provider abs
```

**Reply by rogerhsu1121** [reply]
thanks, it works

**Comment by rogerhsu1121** [comment]
After edit params.yaml and executed
**_microk8s config > kubeconfig
KUBECONFIG=./kubeconfig opctl apply_**

it shows
Starting deployment...

**_2021/04/08 11:05:43 res: namespace/application-system unchanged
customresourcedefinition.apiextensions.k8s.io/applications.app.k8s.io configured
clusterrole.rbac.authorization.k8s.io/application-manager-role configured
clusterrolebinding.rbac.authorization.k8s.io/application-manager-rolebinding configured
service/application-controller-manager-service unchanged
statefulset.apps/application-controller-manager unchanged_**

But I can not access the web page http://localhost:8080, 

Then I open another terminal and check opctl status, it shows connection refused. 
 **sudo opctl app status**
**_Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused_**

How do I resolve this problem?

**Reply by Vafilor** [reply]
In your params.yaml, what did you use for domain/fqdn? It shouldn't be set to localhost.

Here's some values I use for testing:
```yaml
application:
  # First namespace that will be created in Onepanel, more can be added later
  defaultNamespace: onepanel
  domain: "onepanel.test"
  fqdn: "app.onepanel.test"
```

You'll then have to follow the DNS set up starting with

https://docs.onepanel.ai/docs/getting-started/quickstart

Part 1: Step 5
This links you to set up DNS with DNSMasq at https://docs.onepanel.ai/docs/deployment/configuration/dns/

**Reply by rogerhsu1121** [reply]
I follow Part1: Step 5 to install DNSMasq the restart the machine. 
and set my params.yaml

_application:
  defaultNamespace: rogerhsu
  domain: "rogerhsu.com"
  fqdn: "app.rogerhsu.com"
  insecure: true_

then retry 

_microk8s config > kubeconfig
KUBECONFIG=./kubeconfig opctl apply_


but I check by
_opctl app status_

It still shows
_Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused_

**Reply by Vafilor** [reply]
@rogerhsu1121 

It's odd you are seeing that, it might be a permissions issue.

Try exporting the variable for KUBECONFIG like so
```bash
export KUBECONFIG=$(PWD)/kubeconfig
```

then running 
```bash
sudo opctl apply
```

Also, minikube is not supported. You might be able to get it to work, but we've only done thorough testing on microk8s.

If you're still having issues, please check out our slack  - there's a link for it here in the top right https://docs.onepanel.ai/

We might be able to schedule a call or something to walk through the set up.
